src/selectedtests/app/controllers/project_test_mappings_controller.py

```diff
@@ -0,0 +1,71 @@
+"""Controller for the health endpoints."""
+import json
+
+from flask import jsonify, request
+from flask_restplus import abort, Api, fields, Resource
+from evergreen.api import EvergreenApi
+
+from selectedtests.datasource.mongo_wrapper import MongoWrapper
+from selectedtests.app.project_test_mapping_work_item import ProjectTestMappingWorkItem
+from selectedtests.evergreen_helper import get_evg_project
+
+
+def add_project_test_mappings_endpoints(api: Api, mongo: MongoWrapper, evg_api: EvergreenApi):
+    """
+    Add to the given app instance the test mapping jobs endpoints of the service.
+
+    :param api: An instance of a Flask Restplus Api that wraps a Flask instance
+    :param mongo: Mongo Wrapper instance
+    :param evg_api: An instance of the evg_api client
+    """
+    ns = api.namespace("projects", description="Project Test Mappings")
+
+    test_mappings_work_item = ns.model(
+        "TestMappingsWorkItem",
+        {
+            "source_file_regex": fields.String(
+                description="Regex describing folder containing source files in given project",
+                required=True,
+            ),
+            "test_file_regex": fields.String(
+                description="Regex describing folder containing test files in given project.",
+                required=True,
+            ),
+            "module": fields.String(description="Module to include in the analysis"),
+            "module_source_file_regex": fields.String(
+                description="Regex describing folder containing source files in given module"
+            ),
+            "module_test_file_regex": fields.String(
+                description="Regex describing folder containing test files in given module"
+            ),
+        },
+    )
+
+    @ns.route("/<project>/test-mappings")
+    @api.param("project", "The evergreen project identifier")
+    class TestMappingsWorkItem(Resource):
+        @ns.response(200, "Success", test_mappings_work_item)
+        @ns.response(404, "Evergreen project not found", test_mappings_work_item)
+        @ns.response(422, "Work item already exists for project", test_mappings_work_item)
+        @ns.expect(test_mappings_work_item, validate=True)
+        def post(self, project):
+            """Enqueue a project test mapping work item."""
+            evergreen_project = get_evg_project(evg_api, project)
```

Comment:
[nit] It may be helpful to do some additional validation here around the modules - that if a ```module``` field is passed in, ```module_source_file_regex``` and ```module_test_file_regex``` fields are required as well like [here](https://github.com/mongodb/selected-tests/blob/master/src/selectedtests/test_mappings/cli.py#L110). That may help the user with a more friendly error response up front as the asynchronous job work planned makes bubbling errors/issues up to users complicated.

It's hard to show that kind of dependency in swagger but additional text in the descriptions for these fields may help.

Reply:
added to the swagger descriptions